hera_qm/xrfi.py

```diff
@@ -1,51 +1,116 @@
-'''
-Module for all things Radio Frequency Interference Flagging.
-Note these functions currently operate on real numbers only.
-'''
 import numpy as np
 from scipy.signal import medfilt
 
+#############################################################################
+# Functions useful for understanding the statistics in RFI selection/flagging
+#############################################################################
 
 def medmin(d):
-    """Calculate the median minus minimum statistic of array.
-
+    '''Calculate the median minus minimum statistic of array.
     Args:
-        d (array): 2D data array
-
+        d (array): 2D data array of the shape (time,frequency).
     Returns:
         (array): array with the statistic applied.
-    """
+    '''
     mn = np.min(d, axis=0)
     return 2 * np.median(mn) - np.min(mn)
 
 
-def medminfilt(d, K=8):
-    """Filter an array on scales of K indexes with medmin.
-
+def medminfilt(d, Kt=8, Kf=8):
+    '''Filter an array on scales of Kt,Kf indexes with medmin.
     Args:
-        d (array): 2D data array.
-        K (int, optional): integer representing box size to apply statistic.
-
+        d (array): 2D data array of the shape (time,frequency).
+        Kt (int, optional): integer representing box dimension in time to apply statistic.
+        Kf (int, optional): integer representing box dimension in frequency to apply statistic.
     Returns:
         array: filtered array. Same shape as input array.
-    """
+    '''
     d_sm = np.empty_like(d)
     for i in xrange(d.shape[0]):
         for j in xrange(d.shape[1]):
-            i0, j0 = max(0, i - K), max(0, j - K)
-            i1, j1 = min(d.shape[0], i + K), min(d.shape[1], j + K)
+            i0, j0 = max(0, i - Kt), max(0, j - Kf)
+            i1, j1 = min(d.shape[0], i + Kt), min(d.shape[1], j + Kf)
             d_sm[i, j] = medmin(d[i0:i1, j0:j1])
     return d_sm
 
+def detrend_deriv(d, dt=True, df=True):
+    '''XXX This only works ok on sparse RFI.'''
```

Comment:
Please add doc string with description, args, and returns. Here, and throughout, make important notes with: `NB: something something`... `XXX` is (regrettably) often used as a TODO in code, so let's avoid that confusion.